### ISSUE TITLE
False Positive, suggesting unused TemplateHaskell language pragma

### ISSUE THREAD

**Issue by rickowens** [ndmitchell/hlint#1541]
I have a module with the following content:

```
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE TemplateHaskell #-}

module MyModule (
  staticAssets,
) where


import Language.Haskell.TH (Code(Code))
import Network.Wai (Middleware)
import OM.HTTP (staticSite)


staticAssets :: Middleware
staticAssets = $$(Code (staticSite "static"))
```
where OM.HTTP is from the [om-http](https://hackage.haskell.org/package/om-http) package (I don't believe this is relevant, but I can simplify the example further if you believe it is relevant), and the cabal file for this project specifies `Haskell2010`.

hlint 3.6.1 believes that the `TemplateHaskell` pragma is unused and suggests that I should remove it, which is clearly wrong.

```
src/MyModule.hs:2:1-32: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Perhaps you should remove it.

1 hint
```

I suspect it may not recognize the _typed_ splice as being TemplateHaskell?

**Comment by shayne-fletcher** [comment]
is it an instance that https://github.com/ndmitchell/hlint/pull/1535 would fix?

**Comment by rickowens** [comment]
Yes! #1553 does fix this. Apologies, I searched for open issue of this type, but I neglected to search for closed issues.
